src/Autofac/Core/Lifetime/LifetimeScope.cs

```diff
@@ -259,17 +259,8 @@ private ScopeRestrictedRegistry CreateScopeRestrictedRegistry(object tag, Action
             return locals;
         }
 
-        /// <summary>
-        /// Resolve an instance of the provided registration within the context.
-        /// </summary>
-        /// <param name="registration">The registration.</param>
-        /// <param name="parameters">Parameters for the instance.</param>
-        /// <returns>
-        /// The component instance.
-        /// </returns>
-        /// <exception cref="Autofac.Core.Registration.ComponentNotRegisteredException"/>
-        /// <exception cref="DependencyResolutionException"/>
-        public object ResolveComponent(IComponentRegistration registration, IEnumerable<Parameter> parameters)
+        /// <inheritdoc />
+        public object ResolveComponent(Service service, IComponentRegistration registration, IEnumerable<Parameter> parameters)
```

Comment:
This would be
`public object ResolveComponent(ResolutionContext context)`
or something like that.

```c#
public class ResolutionContext
{
  public Service Service { get; set; }
  public IComponentRegistration Registration { get; set; }
  public IEnumerable<Parameter> Parmameters { get; set; }
}
```

Reply:
I think we spoke about the parameter object some time after I created this branch. I definitely prefer that approach, but thought I would start with this to get some feedback before making further changes.